lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2592,7 +2592,7 @@ namespace Js
             return TRUE;
         }
 
-        if (!TaggedNumber::Is(thisInstance))
+        if (!TaggedNumber::Is(instance))
         {
             return JavascriptOperators::SetProperty(RecyclableObject::UnsafeFromVar(thisInstance), RecyclableObject::UnsafeFromVar(instance), propertyId, newValue, info, scriptContext, flags);
```

Comment:
>RecyclableObject::UnsafeFromVar(thisInstance) [](start = 52, length = 45)

Now that !TaggedNumber::Is(thisInstance) is changed does the callers validate that thisInstance is a valid non tagged var?

Reply:
Yes, there's a check for this case in `ProfilingHelpers::ProfiledStFld` which looks like it handles that case. I tried `c.prototype.meth.call(5)` in the example code above and we end up ignoring the number and just passing `instance` itself in the `thisInstance` parameter.